packages/@ember/-internals/runtime/lib/mixins/registry_proxy.ts

```diff
@@ -2,8 +2,35 @@
 @module ember
 */
 
-import { assert } from '@ember/debug';
+import { Registry } from '@ember/-internals/container';
+import { TypeOptions } from '@ember/-internals/container/lib/registry';
 import { Mixin } from '@ember/-internals/metal';
+import { Factory } from '@ember/-internals/owner';
+import { AnyFn } from '@ember/-internals/utils/types';
+import { assert } from '@ember/debug';
+
+export interface IRegistry {
```

Comment:
As above, given the new interface and its apparent relation to `Registry`, can we add a comment explains what differentiates it and how it's expected to be used? 🙏🏼